### ISSUE TITLE
QA Suite - tag push to perf trigger

### ISSUE THREAD

**Issue by cris-oddball** [department-of-veterans-affairs/notification-api#1442]
## User Story - Business Need

The business need for this ticket is to automatically trigger and run a full test suite against the perf environment when a tag is cut against an API branch and then pushed on the trigger, ensuring code reliability before it reaches production.

- [ ] Ticket is understood, and QA has been contacted (if the ticket has a QA label).

### User Story(ies)

**As a**  VANotify engineer
**I want**  the regression to be triggered from a tag pushed to perf, and only run against perf
**So that**  the CI/CD pipeline is fully automated and code is checked for quality

### Additional Info and Resources

When a tag is cut against any API branch and that tag is pushed up to origin master, a tag trigger automatically deploys that code to the perf environment.

The workflow for tag push to perf is in [tag_trigger.yml](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/workflows/tag_trigger.yml)

The regression should **only** be run against perf (since the action only deploys to perf)

## Engineering Checklist

- [x] Update tag_trigger.yml to execute a regression against perf after the code has finished deploying to perf.

## Acceptance Criteria

- [ ] When a tag is cut and pushed for perf, then the regression suite runs against perf after the code is deployed.

## QA Considerations

- QA will need to verify that the full suite runs correctly against the perf environment when a tag is pushed to perf.
- This assumes that the regression suite will be able to validate the code deployed. If API code changes are made that would impact the results of the regression, then a QA branch must be cut and tested against the deployed code on DEV, then merged to QA master before the API tag is pushed to Perf.

**Comment by mjones-oddball** [comment]
Note: Lucas felt this had less complexity and could be a 3 but we went with a 5 to be on the safe side. We should measure and be aware of the end result - did it get completed in less than a 5?

**Comment by mjones-oddball** [comment]
@cris-oddball feels this is a 1 based on it being a repeatable thing and will work this in the current sprint. If there's any snags and she gets stuck it's going back to next up to await Lucas's time/attention.

**Comment by cris-oddball** [comment]
Presently, the [dev_deploy.yml ](https://github.com/department-of-veterans-affairs/notification-api/blob/56286ec6755398bd218016d97949240cfe0f17da/.github/workflows/dev_deploy.yml#L76)workflow runs the regression suite if the environment is Perf or Staging. The code for this is 
```
run-QA-suite:
    needs: [run-deployment]
    if: ${{ (inputs.environment == 'perf') || (inputs.environment == 'staging') }}
    uses: ./.github/workflows/regression-public.yml
    with:
      qa-image-branch: "master"
      environment: "${{ inputs.environment }}"
    secrets: inherit
```

It seems as though this can be modified for use in [tag_trigger.yml](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/workflows/tag_trigger.yml) by adding the code to the end of the file and changing some things
```
run-QA-suite:
    needs: [run-build] # but does it really need this line?
    uses: ./.github/workflows/regression-public.yml
    with:
      qa-image-branch: "master"
      environment: "perf"
    secrets: inherit
```

**Comment by cris-oddball** [comment]
PR approved and merged, tag v1.6.13-tag-trigger-regression has been cut and pushed, and the tag trigger action is running. Awaiting to see if the regression runs as well.

**Comment by cris-oddball** [comment]
The [regression ran](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/6397253084) after the tag was deployed to Perf.

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBc0wrIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--3d84eb9166abd8f12e294ffd75f7babd555e66ca/Screen%20Shot%202023-10-03%20at%201.18.15%20PM.jpg)

Closing ticket as a success.
